tests/repl/test_parser.rs

```diff
@@ -814,6 +814,50 @@ fn let_variable_type_mismatch() -> TestResult {
     fail_test(r#"let x: int = "foo""#, "expected int, found string")
 }
 
+#[test]
+fn let_variable_table_runtime_cast() -> TestResult {
+    // Type::Any should be accepted by compatible types (record can convert to table)
+    run_test(
+        r#"let x: table = ([[a]; [1]] | to nuon | from nuon); $x | describe"#,
+        "table<a: int>",
+    )
+}
+
+#[test]
+fn let_variable_table_runtime_mismatch() -> TestResult {
+    // This conversion should fail due to a key mismatch
+    fail_test(
+        r#"let x: table<b: int> = ([[a]; [1]]  | to nuon | from nuon); $x | describe"#,
+        "can't convert table<a: int> to table<b: int>",
```

Comment:
There should probably be a record to table failure here
```suggestion
        r#"let x: table<a: int> = ({a : 1}  | to nuon | from nuon); $x | describe"#,
        "can't convert record<a: int> to table<a: int>",
```